### PR TITLE
CI: automatic update of nix lock file

### PR DESCRIPTION
**Pull request by imincik**
Regularly update a Nix lock file to get fresh packages from nixpkgs.

Usage of Renovate discussed in https://github.com/OSGeo/grass/pull/3933 .

### CODE REVIEW

File: renovate.json5 [hunk 1]
```diff
@@ -19,5 +19,13 @@
...
-    ]
+    ],
+
+    // update Nix flake.lock file
+    "nix": {
+        "enabled": true
+    },
+    "lockFileMaintenance": {
+        "enabled": true
+    }
```
Comment (echoix) [inline renovate.json5:30]: What about `:maintainLockFilesWeekly` preset instead?

https://docs.renovatebot.com/presets-default/#maintainlockfilesweekly

File: .github/workflows/test-nix.yml [hunk 2]
```diff
@@ -7,6 +7,11 @@ on:
   push:
     tags:
       - '*'
+  pull_request:
+    paths:
+      - 'flake.nix'
+      - 'flake.lock'
+      - 'package.nix'
```
Comment (echoix) [inline .github/workflows/test-nix.yml:14]: You might also want to run when there are changes to that workflow file (that tests nix build).

File: renovate.json5 (hunk 1 above)
Comment (imincik) [reply renovate.json5]: Done, thanks for suggestion.

File: .github/workflows/test-nix.yml (hunk 2 above)
Comment (imincik) [reply .github/workflows/test-nix.yml:14]: Good point, done.

**Review by echoix — APPROVED** [review]

**Pull request merged**